pandas/core/tools/datetimes.py

```diff
@@ -236,40 +236,41 @@ def _convert_listlike_datetimes(arg, box, format, name=None, tz=None,
             require_iso8601 = not infer_datetime_format
             format = None
 
-    try:
-        result = None
+    result = None
+    tz_parsed = None
 
-        if format is not None:
-            # shortcut formatting here
-            if format == '%Y%m%d':
-                try:
-                    result = _attempt_YYYYMMDD(arg, errors=errors)
-                except (ValueError, TypeError, tslibs.OutOfBoundsDatetime):
-                    raise ValueError("cannot convert the input to "
-                                     "'%Y%m%d' date format")
+    if format is not None:
+        # shortcut formatting here
+        if format == '%Y%m%d':
+            try:
+                result = _attempt_YYYYMMDD(arg, errors=errors)
+            except (ValueError, TypeError, tslibs.OutOfBoundsDatetime):
+                raise ValueError("cannot convert the input to "
+                                 "'%Y%m%d' date format")
 
-            # fallback
-            if result is None:
-                try:
-                    result, timezones = array_strptime(
-                        arg, format, exact=exact, errors=errors)
-                    if '%Z' in format or '%z' in format:
-                        return _return_parsed_timezone_results(
-                            result, timezones, box, tz, name)
-                except tslibs.OutOfBoundsDatetime:
+        # fallback
+        if result is None:
+            try:
+                result, timezones = array_strptime(arg, format,
+                                                   exact=exact, errors=errors)
+                if '%Z' in format or '%z' in format:
+                    return _return_parsed_timezone_results(result, timezones,
+                                                           box, tz, name)
+            except tslibs.OutOfBoundsDatetime:
+                if errors == 'raise':
+                    raise
+                result = arg
+            except ValueError:
+                # if format was inferred, try falling back
+                # to array_to_datetime - terminate here
+                # for specified formats
+                if not infer_datetime_format:
                     if errors == 'raise':
                         raise
                     result = arg
-                except ValueError:
-                    # if format was inferred, try falling back
-                    # to array_to_datetime - terminate here
-                    # for specified formats
-                    if not infer_datetime_format:
-                        if errors == 'raise':
-                            raise
-                        result = arg
-
-        if result is None and (format is None or infer_datetime_format):
+
+    if result is None and (format is None or infer_datetime_format):
```

Comment:
Could you comment here why we might hit this branch? It looks like if no format was passed, or a format was passed and it failed?

Reply:
This is probably the branch that gets hit the most often, e.g. `pd.to_datetime([pd.Timestamp.now(), '2015/04/05'])` will go down this route.

Any list/object-dtyped case where a format is not passed.

Any list/object-dtype case where a format is passed, fails, and is allowed to try the fallback (via `infer_datetime_format` kwarg, which defaults to `None`)